lib/linter-rust.coffee

```diff
@@ -273,19 +141,43 @@ class LinterRust
           result.push ['--cfg', "feature=\"#{f}\""]
         result
 
-  ableToJSONErrors: (curDir) =>
-    # current dir is set to handle overrides
+  cachedErrorMode: null
+
+  decideErrorMode: (curDir, commandMode) =>
+    # error mode is cached to avoid delays
+    if @cachedErrorMode? and do @cachedErrorMode.valid
+      Promise.resolve().then () =>
+        do @cachedErrorMode.getResult
+    else
+      # current dir is set to handle overrides
+      atom_linter.exec(@rustcPath, ['--version'], {cwd: curDir}).then (stdout) =>
+        try
+          match = XRegExp.exec(stdout, @patternRustcVersion)
+          if match
+            nightlyWithJSON = match.nightly and match.date > '2016-08-08'
+            stableWithJSON = not match.nightly and semver.gte(match.version, '1.12.0')
+            canUseIntermediateJSON = nightlyWithJSON or stableWithJSON
+            switch commandMode
+              when 'cargo'
+                canUseProperCargoJSON = match.nightly and match.date > '2016-10-10'
```

Comment:
You are right, the right starting version is probably 10-08 or 10-09 since it was merged 15 days ago in  cargo. Thanks for the help, I 'll get this fixed :)